Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/api/ui/TestForApiFragmentViewModel.kt

```diff
@@ -1,20 +1,37 @@
 package de.rki.coronawarnapp.test.api.ui
 
 import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.environment.EnvironmentSetup
 import de.rki.coronawarnapp.storage.LocalData
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
 
-class TestForApiFragmentViewModel @AssistedInject constructor() : CWAViewModel() {
+class TestForApiFragmentViewModel @AssistedInject constructor(
+    private val envSetup: EnvironmentSetup
+) : CWAViewModel() {
 
     val last3HourToggleEvent = SingleLiveEvent<Boolean>()
+    val environmentChangeEvent = SingleLiveEvent<EnvironmentSetup.Type>()
 
     fun setLast3HoursMode(isLast3HoursModeEnabled: Boolean) {
         LocalData.last3HoursMode(isLast3HoursModeEnabled)
         last3HourToggleEvent.postValue(isLast3HoursModeEnabled)
     }
 
+    fun toggleEnvironment(isTestCountyEnabled: Boolean) {
+        if (isTestCountyEnabled) {
```

Comment:
envSetup.currentEnvironment = if(isTestCountyEnabled) alternativeEnvironment else defaultEnvironment

Reply:
adjusted to this. It's the same thing logic wise but saved us few lines 👍